web/profiles/joinup/src/Form/UserCancelForm.php

```diff
@@ -61,45 +60,54 @@ public static function create(ContainerInterface $container) {
    * {@inheritdoc}
    */
   public function buildForm(array $form, FormStateInterface $form_state) {
-    // Prepare a list of collections where the user is the sole owner.
-    $collections = $this->relationManager->getCollectionsWhereSoleOwner($this->entity);
+    // Prepare a list of groups where the user is the sole owner.
+    $groups = $this->relationManager->getGroupsWhereSoleOwner($this->entity);
+    $group_data = [];
+
+    if (!empty($groups)) {
+      foreach ($groups as $group) {
+        $group_data[$group->bundle()][] = $group->toLink($group->label());
+      }
+
+      $rdf_storage = $this->entityTypeManager->getStorage('rdf_type');
+      foreach (['collection', 'solution'] as $bundle) {
+        $bundle_type = $rdf_storage->load($bundle);
+        if (!empty($group_data[$bundle])) {
+          $form[$bundle] = [
+            '#theme' => 'item_list',
+            '#items' => $group_data[$bundle],
+            '#title' => $bundle_type->getCountLabel(count($group_data[$bundle])),
+            '#weight' => 0,
+          ];
+        }
+      }
```

Comment:
Let's move this part in a trait, as is duplicated code, and reuse it in both: `UserMultipleCancelConfirm` and `UserCancelForm`.

Reply:
The code is slightly different in terms that it has different depth within the form array (the first is `$form[$bundle']` and the second one is `$form[$account->id()]['bundle']`.
Also, it would be weird to have a method trait of 15 lines have entries added to the final build array.
Can we skip this one?

Reply:
Yesh, I agee.